src/main/java/donnafin/storage/JsonAdaptedPerson.java

```diff
@@ -140,10 +120,45 @@ public Person toModelType() throws IllegalValueException {
         }
         final Address modelAddress = new Address(address);
 
+        if (notes == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Notes.class.getSimpleName()));
+        }
         final Notes modelNotes = new Notes(notes);
-        final Set<Policy> modelPolicies = new HashSet<>(personPolicies);
-        final Set<Asset> modelAssets = new HashSet<>(personAssets);
-        final Set<Liability> modelLiabilities = new HashSet<>(personLiabilities);
+
+        final Set<Policy> modelPolicies = new HashSet<>();
+        final Set<Asset> modelAssets = new HashSet<>();
+        final Set<Liability> modelLiabilities = new HashSet<>();
+
+        if (policies == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, "policies"));
+        }
+        for (JsonAdaptedPolicy policy : policies) {
+            modelPolicies.add(policy.toModelType());
+        }
+        if (policies.size() != modelPolicies.size()) {
+            throw new IllegalValueException(String.format(DUPLICATE_ATTRIBUTE_IN_LIST, "policies"));
+        }
+
+        if (assets == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, "assets"));
+        }
+        for (JsonAdaptedAsset asset : assets) {
+            modelAssets.add(asset.toModelType());
+        }
+        if (assets.size() != modelAssets.size()) {
```

Comment:
Is this ever possible?

Reply:
same for the rest

Reply:
hashset removes duplicates. this is to prevent users from inserting duplicates from the json

Reply:
okay niceu